src/ray/common/scheduling/cluster_resource_scheduler.cc

```diff
@@ -19,6 +50,87 @@ std::string NodeResources::DebugString() {
   return buffer.str();
 }
 
+std::string PrintVector(std::vector<double> &vector) {
+  std::stringstream buffer;
+
+  buffer << "[";
+  for (size_t i = 0; i < vector.size(); i++) {
+    buffer << vector[i];
+    if (i < vector.size() - 1) {
+      buffer << ", ";
+    }
+  }
+  buffer << "]";
+  return buffer.str();
+}
+
+bool NodeResourceInstances::operator==(const NodeResourceInstances &other) {
+  for (size_t i = 0; i < PredefinedResources_MAX; i++) {
+    if (!EqualVectors(this->predefined_resources[i].total,
+                      other.predefined_resources[i].total)) {
+      return false;
+    }
+    if (!EqualVectors(this->predefined_resources[i].available,
+                      other.predefined_resources[i].available)) {
+      return false;
+    }
+  }
+
+  if (this->custom_resources.size() != other.custom_resources.size()) {
+    return true;
```

Comment:
Shouldn't this be `return false`?

Reply:
Right again! (BTW, bot this and the previous similar mistake occurs in methods that have been only used for testing ;-)